File: client/src/app/invm.ts
import { enums } from './enums';

export namespace invm { 
	/* invm -  История */ 

 export interface   invm_info { // История
	invm_infoId:string; // Primary key field
	fromcell:string; //Из ячейки -> invwh_cell
	toCell:string; //В ячейку -> invwh_cell
	theDep:string; //В отдел -> invd_dep
	storepartid:string; //Запчасть -> invp_data
	qty:Number; // Количество
	theOP:string; //Операция -> invd_op
	optime:string;  // Время операции
	theUser:string; //Оператор -> xUserInfo
	// add dereference fields 
	fromcell_name :string; // dereference for invwh_cell
	toCell_name :string; // dereference for invwh_cell
	theDep_name :string; // dereference for invd_dep
	storepartid_name :string; // dereference for invp_data
	theOP_name :string; // dereference for invd_op
	theUser_name :string; // dereference for xUserInfo
 }
}

File: server/DataCore/invm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace inv19.models { 
	/* invm -  История */ 

 public class  invm_info { // История
	 public System.Guid  invm_infoId{ get; set; } // Идентификатор (первичный ключ)
	public System.Guid  fromcell { get; set; } //Из ячейки
	
	public System.Guid  toCell { get; set; } //В ячейку

	public System.Guid  theDep { get; set; } //В отдел
	
	public System.Guid  storepartid { get; set; } //Запчасть
	
	[Required]
	public double  Qty{ get; set; } // Количество
	public System.Guid  theOP { get; set; } //Операция
	
	[Required]
	public DateTime  optime{ get; set; } // Время операции
	public System.Guid  theUser { get; set; } //Оператор
	
 }
}